pkg/planner/core/task.go

```diff
@@ -1685,17 +1686,17 @@ func (p *basePhysicalAgg) convertAvgForMPP() *PhysicalProjection {
 			// insert a sum(column)
 			avgSum := aggFunc.Clone()
 			avgSum.Name = ast.AggFuncSum
-			avgSum.TypeInfer4AvgSum(avgSum.RetTp)
+			avgSum.TypeInfer4AvgSum(true, aggFunc.Args[0].GetType(exprCtx.GetEvalCtx()))
```

Comment:
```suggestion
			avgSum.TypeInfer4AvgSum(true, aggFunc.Args[0].GetType(exprCtx.GetEvalCtx()))
```

This kind of multiple calls in one line will be very uncomfortable in the debugging stage. It is recommended to split it